backend/clubs/management/commands/stress_test.py

```diff
@@ -0,0 +1,173 @@
+        self.num_questions_per_club = 2
+        self.total_submissions = 1
+        self.club_prefix = "test_club_"
+        self.user_prefix = "test_user_"
+
+        self.uri = "/users/question_response/"
+        self.factory = APIRequestFactory()
+        self.view = UserViewSet.as_view({"post": "question_response"})
+
+        self.club_question_ids = {}
+        self.users = []
+        now = timezone.now()
+
+        # Create Clubs
+        clubs = [
+            Club(code=(self.club_prefix + str(i)), name=(f"Test Club {i}"))
+            for i in range(self.num_clubs)
+        ]
+        Club.objects.bulk_create(clubs)
+        clubs = Club.objects.filter(code__startswith=self.club_prefix)
+
+        # Create Club Applications
+        applications = [
+            ClubApplication(
+                name="Test Application",
+                club=club,
+                application_start_time=now - datetime.timedelta(days=1),
+                application_end_time=now + datetime.timedelta(days=3),
+                result_release_time=now + datetime.timedelta(weeks=1),
+                external_url="https://pennlabs.org/",
+            )
+            for club in clubs
+        ]
+        ClubApplication.objects.bulk_create(applications)
+        applications = ClubApplication.objects.filter(club__in=clubs)
+
+        # Create Club Application Questions
+        questions = [
+            ApplicationQuestion(
+                question_type=ApplicationQuestion.FREE_RESPONSE,
+                prompt="Answer the prompt you selected",
+                word_limit=150,
+                application=application,
+            )
+            for _ in range(self.num_questions_per_club)
+            for application in applications
+        ]
+        ApplicationQuestion.objects.bulk_create(questions)
+
+        clubs_data = Club.objects.filter(
+            code__startswith=self.club_prefix
+        ).prefetch_related(
+            Prefetch(
+                "clubapplication_set",
+                queryset=ClubApplication.objects.prefetch_related("questions"),
+            )
+        )
+        for club in clubs_data:
+            question_ids = [
+                str(question.id)
+                for application in club.clubapplication_set.all()
+                for question in application.questions.all()
+            ]
+            self.club_question_ids[club.id] = question_ids
+        print("Finished setting up clubs.")
+
+        # Create Users (Applicants)
+        User = get_user_model()
+        User.objects.bulk_create(
+            [
+                User(
+                    username=self.user_prefix + str(i),
+                    email=str(i) + "@upenn.edu",
+                    password="test",
+                )
+                for i in range(self.num_users)
+            ]
+        )
+        self.users = list(User.objects.filter(username__startswith=self.user_prefix))
+        print("Finished setting up users.")
+
+    @sync_to_async
```

Comment:
This decorator is misleading. All this is doing is taking your function and convering it to an `awaitable`. It unfortunately doesn't make your code "async for free". In fact, with `thread_sensitive = True` (which is the default behaviour), these functions will all run in the main thread, and probably even slower then usual since it goes through the overhead of "asyncifying". Some more information [here](https://docs.djangoproject.com/en/5.0/topics/async/#asgiref.sync.sync_to_async)

It isn't worth changing it back since this is just a test but I thought I should let you know about this behaviour. If you wanted better async, you might want to use httpx or something like that (I think newer versions of Django support Async more natively).

Reply:
Yeah I was starting to notice that in the end. I originally was designing it as sending HTTP requests and was going to use `httpx` but then the mocking seemed like a better option. Thanks for sharing!